### PR TITLE
feat: ext argo app rbac and missing common features

### PR DESCRIPTION
**Pull request by RajeevRanjan27**
# Description

Fixes https://github.com/devtron-labs/devtron/issues/5527

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [x] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: pkg/argoApplication/ArgoApplicationService.go [hunk 1]
```diff
@@ -450,3 +455,43 @@ func (impl *ArgoApplicationServiceImpl) GetRestConfigForExternalArgo(ctx context
 	}
 	return restConfig, nil
 }
+
+func (impl *ArgoApplicationServiceImpl) HibernateArgoApplication(ctx context.Context, app *bean.ArgoAppIdentifier, hibernateRequest *openapi.HibernateRequest) ([]*openapi.HibernateStatus, error) {
+	_, clusterBean, _, err := impl.GetClusterConfigFromAllClusters(app.ClusterId)
+	//conf, err := impl.helmAppService.GetClusterConf(app.ClusterId)
```
Comment (prakash100198) [inline pkg/argoApplication/ArgoApplicationService.go:461]: remove

File: pkg/argoApplication/ArgoApplicationService.go [hunk 2]
```diff
@@ -450,3 +455,43 @@ func (impl *ArgoApplicationServiceImpl) GetRestConfigForExternalArgo(ctx context
...
+		impl.logger.Errorw("HibernateArgoApplication", "error in getting the cluster config", err, "clusterId", app.ClusterId)
+		return nil, err
+	}
+	conf := ConvertClusterBeanToGrpcConfig(clusterBean)
+
+	req := service.HibernateReqAdaptor(hibernateRequest)
+	req.ClusterConfig = conf
+	res, err := impl.helmAppClient.Hibernate(ctx, req)
+	if err != nil {
+		impl.logger.Errorw("HibernateArgoApplication", "error in hibernating the requested resource", err, "clusterId", app.ClusterId)
```
Comment (prakash100198) [inline pkg/argoApplication/ArgoApplicationService.go:472]: log identifier name for better debugging

File: pkg/argoApplication/ArgoApplicationService.go [hunk 3]
```diff
@@ -450,3 +455,43 @@ func (impl *ArgoApplicationServiceImpl) GetRestConfigForExternalArgo(ctx context
...
+		return nil, err
+	}
+	response := service.HibernateResponseAdaptor(res.Status)
+	return response, nil
+}
+
+func (impl *ArgoApplicationServiceImpl) UnHibernateArgoApplication(ctx context.Context, app *bean.ArgoAppIdentifier, hibernateRequest *openapi.HibernateRequest) ([]*openapi.HibernateStatus, error) {
+	_, clusterBean, _, err := impl.GetClusterConfigFromAllClusters(app.ClusterId)
+	//conf, err := impl.helmAppService.GetClusterConf(app.ClusterId)
```
Comment (prakash100198) [inline pkg/argoApplication/ArgoApplicationService.go:481]: remove

File: pkg/fluxApplication/FluxApplicationService.go [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+	conf, err := impl.helmAppService.GetClusterConf(app.ClusterId)
+	if err != nil {
+		impl.logger.Errorw("HibernateFluxApplication", "error in getting the cluster config", err, "clusterId", app.ClusterId)
+		return nil, err
+	}
+	req := service.HibernateReqAdaptor(hibernateRequest)
+	req.ClusterConfig = conf
+	res, err := impl.helmAppClient.Hibernate(ctx, req)
+	if err != nil {
+		impl.logger.Errorw("HibernateFluxApplication", "error in hibernating the requested resource", err, "clusterId", app.ClusterId)
```
Comment (prakash100198) [inline pkg/fluxApplication/FluxApplicationService.go:57]: log app name

**Review by prakash100198 — CHANGES_REQUESTED** [review]

**Review by prakash100198 — APPROVED** [review]

**Review by nishant-d — APPROVED** [review]

**Review by vikramdevtron — APPROVED** [review]

**Pull request merged**